misc/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -51,6 +54,10 @@ def open(cls, connection):
                 f"  Supported versions: {SUPPORTED_MATERIALIZE_VERSIONS}"
             )
 
+        creds = connection.credentials
+        if creds.cluster:
+            cursor.execute("SET cluster = '{}'".format(creds.cluster))
```

Comment:
This doesn't look safe wrt cluster names that contain `'`. Edge case but probably worth getting right. I think `cursor.execute("SET CLUSTER = %s", creds.cluster)` is what you want?